### PR TITLE
Workers and data URLs

### PR DESCRIPTION
**Pull request by annevk**
Provide basic tests for https://github.com/whatwg/html/pull/1782.

### CODE REVIEW

File: workers/data-url-shared.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<title>data URL shared worker</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id=log></div>
+<script>
+var worker = new SharedWorker("data:,onconnect = (e) => { fetch('/').then(() => e.ports[0].postMessage('fail'), () => e.ports[0].postMessage('pass')) }") // not same-origin
```
Comment (foolip) [inline workers/data-url-shared.html]: https://fetch.spec.whatwg.org/ says that HTML "prevents creation of shared workers from such URLs." but that doesn't seem to match https://html.spec.whatwg.org/#dom-sharedworker

If they shouldn't be created at all, then presumably the constructor should throw?

File: workers/data-url.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<title>data URL worker</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id=log></div>
+<script>
+var worker = new Worker("data:,fetch('/').then(() => self.postMessage('fail'), () => self.postMessage('pass'))") // not same-origin
+worker.onmessage = (e) => {
+  assert_equals(e.data, "pass")
```
Comment (foolip) [inline workers/data-url.html]: testharness.js does implicitly create an `async_test` here, but for some reason a failure is still a harness error. @jgraham, how do these implicitly created tests work?

**Review by foolip — CHANGES_REQUESTED** [review]
I've tested to run these:
https://w3c-test.org/submissions/3943/workers/data-url.html
https://w3c-test.org/submissions/3943/workers/data-url-shared.html

Only Firefox passes, the rest fail with a harness error.

File: workers/data-url-shared.html (hunk 1 above)
Comment (annevk) [inline workers/data-url-shared.html]: Fetch needs to be updated, since we decided to change how shared workers should work.

Comment (annevk) [inline workers/data-url-shared.html]: I updated Fetch.

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
Other now fail in the right way, merging.

**Comment by zcorpan** [comment]
These already tested data: URLs

workers/constructors/SharedWorker/same-origin.html
workers/constructors/Worker/same-origin.htm

**Comment by annevk** [comment]
Hmm, I remember running a grep looking for "data:". Weird. But those tests are not testing the origin of the worker itself, so I guess it was still useful for me to write those tests.

**Pull request merged**